### PR TITLE
Ensure providerless kubelet does not pull cloud providers

### PR DESCRIPTION
**Pull request by dims**
- `providerless` tag is still pulling in AWS/Azure/google cloud providers when building kubelet.
- updated verify scripts to ensure we don't regress

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

/area code-organization

#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
Verified by hand using:
```
strings _output/local/go/bin/kubelet | grep -e Azure/azure-sdk-for-go -e github.com/aws/aws-sdk-go -e google.golang.org/api
```

Updated the `hack/verify-typecheck-*.sh` scripts to do a similar check.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
cc @adisky

**Review by troy0820 — COMMENTED** [review]
/lgtm

File: pkg/kubelet/cadvisor/cadvisor_linux.go [hunk 1]
```diff
@@ -31,12 +31,6 @@ import (
 	_ "github.com/google/cadvisor/container/crio/install"
 	_ "github.com/google/cadvisor/container/systemd/install"
 
-	// Register cloud info providers.
-	// TODO(#68522): Remove this in 1.20+ once the cAdvisor endpoints are removed.
-	_ "github.com/google/cadvisor/utils/cloudinfo/aws"
-	_ "github.com/google/cadvisor/utils/cloudinfo/azure"
-	_ "github.com/google/cadvisor/utils/cloudinfo/gce"
```
Comment (liggitt) [inline pkg/kubelet/cadvisor/cadvisor_linux.go:38]: https://github.com/kubernetes/kubernetes/issues/68522#issuecomment-724928827 doesn't look like it is completed yet... is it?

Comment (liggitt) [reply pkg/kubelet/cadvisor/cadvisor_linux.go:38]: oh, nm... I see this is just moving files

File: hack/verify-typecheck-providerless.sh [hunk 2]
```diff
@@ -27,3 +27,11 @@ cd "${KUBE_ROOT}"
 # verify the providerless build
 # https://github.com/kubernetes/enhancements/blob/master/keps/sig-cloud-provider/1179-building-without-in-tree-providers/README.md
 hack/verify-typecheck.sh --skip-test --tags=providerless --ignore-dirs=test
+
+# verify using go list
+if _out="$(go list -mod=readonly -tags "providerless" -e -json  k8s.io/kubernetes/cmd/kubelet/... \
```
Comment (BenTheElder) [inline hack/verify-typecheck-providerless.sh:32]: only kubelet?

Comment (dims) [reply hack/verify-typecheck-providerless.sh:32]: Let's see if this holds up before adding more in 1.22 :)

Comment (dims) [reply hack/verify-typecheck-providerless.sh:32]: Ideally I want to add a restrict regex in the typecheck itself

**Comment by dims** [comment]
/hold

For release lead @palnabarun 

/milestone v1.21

**Comment by palnabarun** [comment]
/hold cancel

Doesn't look like this may cause any issues.

**Comment by BenTheElder** [comment]
Known flake probing container test
/retest

**Pull request merged**